js/damage_calc.js

```diff
@@ -173,6 +176,125 @@ function calculateSpellDamage(stats, weapon, conversions, use_spell_damage, igno
+    spell_type:     str             [TODO: DEPRECATED/REMOVE] "healing" or "damage"
+    scaling:        Optional[str]   [DEFAULT: "spell"] "melee" or "spell"
+    use_atkspd:     Optional[bool]  [DEFAULT: true] true to factor attack speed, false otherwise.
+    display:        Optional[str]   [DEFAULT: "total"] "total" to sum all parts. Or, the name of a spell part
+    parts:          List[part]      Parts of this spell (different stuff the spell does basically)
+}
+
+NOTE: when using `replace_spell` on an existing spell, all fields become optional.
+Specified fields overwrite existing fields; unspecified fields are left unchanged.
+
+
+There are three possible spell "part" types: damage, heal, and total.
+
+part: spell_damage | spell_heal | spell_total
+
+spell_damage: {
+    name:           str != "total"  Name of the part.
+    type:           "damage"        [TODO: DEPRECATED/REMOVE] flag signaling what type of part it is. Can infer from fields
+    multipliers:    array[num, 6]   floating point spellmults (though supposedly wynn only supports integer mults)
+}
+spell_heal: {
+    name:           str != "total"  Name of the part.
+    type:           "heal"          [TODO: DEPRECATED/REMOVE] flag signaling what type of part it is. Can infer from fields
+    power:          num             floating point healing power (1 is 100% of max hp).
+}
+spell_total: {
+    name:           str != "total"  Name of the part.
+    type:           "total"         [TODO: DEPRECATED/REMOVE] flag signaling what type of part it is. Can infer from fields
+    hits:           Map[str, num]   Keys are other part names, numbers are the multipliers. Undefined behavior if subparts
+                                        are not the same type of spell. Can only pull from spells defined before it.
+}
+
+
+Before passing to display, use the following structs.
+NOTE: total is collapsed into damage or healing.
+
+spell_damage: {
+    type:           "damage"        Internal use
+    name:           str             Display name of part. Should be human readable
+    normal_min:     array[num, 6]   floating point damages (no crit, min), can be less than zero. Order: NETWFA
+    normal_max:     array[num, 6]   floating point damages (no crit, max)
+    normal_total:   array[num, 2]   (min, max) noncrit total damage (not negative)
+    crit_min:       array[num, 6]   floating point damages (crit, min), can be less than zero. Order: NETWFA
+    crit_max:       array[num, 6]   floating point damages (crit, max)
+    crit_total:     array[num, 2]   (min, max) crit total damage (not negative)
+}
+spell_heal: {
+    type:           "heal"          Internal use
+    name:           str             Display name of part. Should be human readable
+    heal_amount:    num             floating point HP healed (self)
+}
+
+*/
+
+const default_spells = {
+    wand: [{
+        name: "Magic Strike",  // TODO: name for melee attacks?
```

Comment:
what

Reply:
are these just for flavortext when we display them? I'd probably just go "Wand Melee" or "Mage Melee" so we avoid confusion with actual Wynn names

Reply:
fair